rewrite-java/src/main/java/org/openrewrite/java/CreateJavaClass.java

```diff
@@ -0,0 +1,156 @@
+            @JsonProperty("className") String className,
+            @Language("java") @JsonProperty("classTemplate") String classTemplate,
+            @JsonProperty("overwriteExisting") @Nullable Boolean overwriteExisting
+    ) {
+        this.sourceRoot = sourceRoot;
+        this.packageName = packageName;
+        this.className = className;
+        this.classTemplate = classTemplate;
+        this.overwriteExisting = overwriteExisting;
+    }
+
+    @Override
+    public @NonNull String getDisplayName() {
+        return "Create Java class";
+    }
+
+    @Override
+    public @NonNull String getDescription() {
+        return "Create a new Java class.";
+    }
+
+    @Override
+    public @NonNull AtomicBoolean getInitialValue(@NonNull ExecutionContext ctx) {
+        return new AtomicBoolean(true);
+    }
+
+    @Override
+    public @NonNull TreeVisitor<?, ExecutionContext> getScanner(@NonNull AtomicBoolean shouldCreate) {
+        return new CreateFileVisitor(getSourcePath(), shouldCreate);
+    }
+
+    @Override
+    public @NonNull Collection<SourceFile> generate(AtomicBoolean shouldCreate, @NonNull ExecutionContext ctx) {
+        if (shouldCreate.get()) {
+            String classTemplate = StringUtils.isBlank(this.classTemplate) ? NEW_CLASS_TEMPLATE : this.classTemplate;
+            return JavaParser.fromJavaVersion().build().parse(String.format(classTemplate, packageName, className))
+                    .map(brandNewFile -> (SourceFile) brandNewFile.withSourcePath(Paths.get(getSourcePath())))
+                    .collect(Collectors.toList());
+        }
+        return emptyList();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(AtomicBoolean created) {
+        Path path = Paths.get(getSourcePath());
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+                if ((created.get() || Boolean.TRUE.equals(overwriteExisting)) && path.toString().equals(cu.getSourcePath().toString())) {
+                    J.CompilationUnit newCu = JavaTemplate.builder(String.format(classTemplate, packageName, className))
+                            .contextSensitive()
+                            .doBeforeParseTemplate(System.out::println)
+                            .build()
+                            .apply(getCursor(), new JavaCoordinates(cu, Space.Location.ANY, JavaCoordinates.Mode.REPLACEMENT, null))
+                            .withMarkers(cu.getMarkers())
+                            .withId(cu.getId());
+
+                    return newCu.withSourcePath(path);
+                }
+                return cu;
+            }
+        };
+    }
+
+    private @NonNull String getSourcePath() {
+        return String.format("src/%s/java/%s/%s.java", sourceRoot, packageName.replace('.', '/'), className);
```

Comment:
It may be better to just allow the recipe to specify the source path itself as an input option. As written presently, this would only be able to add a Java class file to the root project.